neuro_san/service/mcp/processors/mcp_tools_processor.py

```diff
@@ -0,0 +1,155 @@
+        :return: json dictionary with tool response in MCP format;
+                 or json dictionary with error message in MCP format.
+        """
+
+        service_provider: AsyncAgentServiceProvider = self.agent_policy.allow(tool_name)
+        if service_provider is None:
+            # No such tool is found:
+            return McpErrorsUtil.get_tool_error(request_id, f"Tool not found: {tool_name}")
+        service: AsyncAgentService = service_provider.get_service()
+
+        input_request: Dict[str, Any] = self._get_chat_input_request(prompt)
+        response_text: str = ""
+        try:
+            result_generator = service.streaming_chat(input_request, metadata)
+            async for result_dict in result_generator:
+                partial_response: str = await self._extract_tool_response_part(result_dict)
+                if partial_response is not None:
+                    response_text = response_text + partial_response
+        except Exception as exc:  # pylint: disable=broad-exception-caught
+            self.logger.error(metadata, "Tool %s execution failed: %s", tool_name, str(exc))
+            return McpErrorsUtil.get_tool_error(request_id, f"Failed to execute tool {tool_name}")
+
+        # Return tool call result:
+        return {
+            "jsonrpc": "2.0",
+            "id": RequestsUtil.safe_request_id(request_id),
+            "result": {
+                "content": [
+                    {
+                        "type": "text",
+                        "text": RequestsUtil.safe_message(response_text)
+                    }
+                ],
+                "isError": False
+            }
+        }
+
+    async def _get_tool_description(self, agent_name: str, metadata: Dict[str, Any]) -> Dict[str, Any]:
+        service_provider: AsyncAgentServiceProvider = self.agent_policy.allow(agent_name)
+        if service_provider is None:
+            return None
+        service: AsyncAgentService = service_provider.get_service()
+        function_dict: Dict[str, Any] = await service.function({}, metadata)
+        tool_description: str = function_dict.get("function", {}).get("description", "")
+        return {
+            "name": agent_name,
+            "description": tool_description,
+            "inputSchema": {
+                "type": "object",
+                "properties": {
+                    "input": {
+                        "type": "string",
+                        "description": "text input for chat request"
+                    }
+                },
+                "required": ["input"]
+            }
+        }
+
+    async def _extract_tool_response_part(self, response_dict: Dict[str, Any]) -> str:
+        response_part_dict: Dict[str, Any] = response_dict.get("response", {})
+        response_type: str = response_part_dict.get("type", "")
+        if response_type == "AGENT_FRAMEWORK":
+            return response_part_dict.get("text", None)
```

Comment:
This is OK for getting things going, but please file a ticket to be sure the structure information gets in here as well.  See BasicMessageProcessor.get_compiled_answer()

Also separate future work would involve allowing specific sly_data fields through. Nothing comes through by default, so you have that covered already.  Will need to think on best place in hocon to do that.